docs/DeveloperGuide.md

```diff
@@ -1,38 +1,201 @@
+
+![load Chassis Sequential Diagram](uml-pictures/loadChassisSequential.png)
+
+### Parser class
+The UML class diagram shows the features of the Parser class:
+
+![Parser Class Diagram](uml-pictures/ParserClass.png)
+
+The purpose of the Parser class is to parse the user input into commands for execution. With reference to the diagram,
+the Parser class does not have any attributes and only contains public methods for parsing of the user input.
+The methods in the Parser class creates a subclass which inherits from the abstract superclass Command class. 
+From the diagram, Command class has a private attribute named arguments which stores the user inputs as a String. The Command class
+contains methods that help to set and retrieve the private attribute in the class. 
+Overall, this class diagram provides an overview of the purpose of the Parser class.
+
+####  ViewerAddCommand() method
+This Sequential analysis will show how the ViewerAddCommand() method works, this will also serve as an example for the ViewerDeleteCommand and ViewerViewCommand.
+During the ParseCommand, when a add command is detected in the user input, it will call the `ViewerAddCommand` method with 
+a name as its argument. The `ViewerAddCommand` then calls a method to retrieve the argument.The `ViewerAddCommand`
+checks if the argument is empty and throws an `PPException` if it is empty. A `PC` object is created with the argument as
+an input. The `add` method is called by accessing the `dataStorage` and `pcList` classes to add the `PC` object to the list.
+Finally, a message is returned to inform the user that the `ViewerAddCommand` method has been executed successfully.
+
+A UML sequence diagram showing the interactions between the different objects involved in handling this method can be
+found below:
+![Viewer Add Command Sequential Diagram](uml-pictures/ViewerAddCommand.png)
+### PC Class Architecture
+
+![PC Class Diagram](uml-pictures/PcDiagram.png)
+
+This UML class diagram shows the classes and their relationships in a computer system, which is represented by the `PC` class. The PC class has several attributes, including name, isPrebuilt, cpu, cpuCooler, gpu, motherboard, ram, storage, psu, chassis, and budget.
+
+The `CPU`, `CPUCooler`, `GPU`, `Motherboard`, `RAM`, `Chassis`, `PSU` and `Storage` classes are components that can be used to build a `PC` object. Each component class has its own attributes and methods specific to that component.
+
+The `Component` class is an abstract class that represents a generic computer component. It has attributes such as name, brand, and price, as well as methods to get and set those attributes.
+
+The `PC` class has methods to set and get the components of a `PC` object, as well as methods to set and get the name, isPrebuilt, and budget attributes.
+
+Overall, this class diagram provides a high-level overview of the components that make up a computer system and their relationships to each other.
+
+### Command class
```

Comment:
Since the developer guide provides a comprehensive diagram of list Command, it is advised to add on other command methods analysis to be consistent. Very informative diagram.